src/BugsnagServiceProvider.php

```diff
@@ -358,8 +358,17 @@ protected function setupPaths(Client $client, $base, $path, $strip, $project)
             return;
         }
 
-        $client->setStripPath($base);
-        $client->setProjectRoot($path);
+        if (isset($stripRegex)) {
```

Comment:
`bugsnag-php` calls setStringPathRegex whenever `setProjectRootRegex` is called anyway, so it gets rather confusing as to which option will actually get set last. Also, the call to setStripPath above (L353) is always going to be overwritten from what I can see.

Instead, can we look to simplify by setting these fields in the following order starting with the most general?

1. Set Project Root = $app->path()
2. Set Strip Path = $app->basePath()
3. If Project Regex is given, set Project Regex
4. If Strip Regex is given, set Strip Regex
5. If Project Root is given, set Project Root
6. If Strip Path is given, set Strip Path (and Project Root if neither project regex or path are set)

In this order, the strip paths will be set before project roots (and so not get overwritten) and the regex's will be set before absolute paths (and so the most specific value is used).

Reply:
Hey @tomlongridge thank you very much for your feedback!

I've submitted some few changes to adopt the PR. I've only changed hierarchy giving regex parameters more weight as in my opinion is fewer general than using project or strip paths

Let me know what do you think ;)

Reply:
Looks good and I've had chance to test it out. Just a couple of further comments.